extensions/sql-database-projects/src/models/connections/connectionService.ts

```diff
@@ -0,0 +1,171 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import * as constants from '../../common/constants';
+import * as utils from '../../common/utils';
+import * as vscode from 'vscode';
+import { ConnectionResult } from 'azdata';
+import { IFireWallRuleError } from 'vscode-mssql';
+
+export interface ISqlConnectionProperties {
+	tenantId?: string,
+	accountId?: string
+	serverName: string,
+	userName: string,
+	password: string,
+	port: number,
+	dbName: string,
+	profileName?: string,
+	connectionRetryTimeout?: number
+}
+
+/**
+ * Includes methods to open connections and interact with connection views
+ */
+export class ConnectionService {
+
+	constructor(private _outputChannel: vscode.OutputChannel) {
+	}
+
+	private DefaultSqlRetryTimeoutInSec: number = 10;
+	private DefaultSqlNumberOfRetries: number = 3;
+
+	// Connects to a database
+	private async connectToDatabase(profile: ISqlConnectionProperties, saveConnectionAndPassword: boolean, database: string): Promise<ConnectionResult | string | undefined> {
+		const getAzdataApi = await utils.getAzdataApi();
+		const vscodeMssqlApi = getAzdataApi ? undefined : await utils.getVscodeMssqlApi();
+		if (getAzdataApi) {
+			const connectionProfile = {
+				password: profile.password,
+				serverName: `${profile.serverName},${profile.port}`,
+				database: database,
+				savePassword: saveConnectionAndPassword,
+				userName: profile.userName,
+				providerName: 'MSSQL',
+				saveProfile: false,
+				id: '',
+				connectionName: profile.profileName,
+				options: [],
+				authenticationType: 'SqlLogin'
+			};
+			return await getAzdataApi.connection.connect(connectionProfile, saveConnectionAndPassword, false);
+		} else if (vscodeMssqlApi) {
+			const connectionProfile = {
+				password: profile.password,
+				server: `${profile.serverName}`,
+				port: profile.port,
+				database: database,
+				savePassword: saveConnectionAndPassword,
+				user: profile.userName,
+				authenticationType: 'SqlLogin',
+				encrypt: false,
+				connectTimeout: 30,
+				applicationName: 'SQL Database Project',
+				accountId: profile.accountId,
+				azureAccountToken: undefined,
+				applicationIntent: undefined,
+				attachDbFilename: undefined,
+				connectRetryCount: undefined,
+				connectRetryInterval: undefined,
+				connectionString: undefined,
+				currentLanguage: undefined,
+				email: undefined,
+				failoverPartner: undefined,
+				loadBalanceTimeout: undefined,
+				maxPoolSize: undefined,
+				minPoolSize: undefined,
+				multiSubnetFailover: undefined,
+				multipleActiveResultSets: undefined,
+				packetSize: undefined,
+				persistSecurityInfo: undefined,
+				pooling: undefined,
+				replication: undefined,
+				trustServerCertificate: undefined,
+				typeSystemVersion: undefined,
+				workstationId: undefined,
+				profileName: profile.profileName,
+				expiresOn: undefined,
+				tenantId: profile.tenantId
```

Comment:
wow this is a lot more parameters needed for the vscode-mssql connection. Not in this PR, but it would be nice to see if this can be updated in the future so that most of these are optional so they don't need to be set to undefined.